Sources/Foundation/JSONSerialization.swift

```diff
@@ -549,523 +615,913 @@ private struct JSONWriter {
 }
 
 //MARK: - JSONDeserializer
-private struct JSONReader {
-
-    static let whitespaceASCII: [UInt8] = [
-        0x09, // Horizontal tab
-        0x0A, // Line feed or New line
-        0x0D, // Carriage return
-        0x20, // Space
-    ]
-
-    struct Structure {
-        static let BeginArray: UInt8     = 0x5B // [
-        static let EndArray: UInt8       = 0x5D // ]
-        static let BeginObject: UInt8    = 0x7B // {
-        static let EndObject: UInt8      = 0x7D // }
-        static let NameSeparator: UInt8  = 0x3A // :
-        static let ValueSeparator: UInt8 = 0x2C // ,
-        static let QuotationMark: UInt8  = 0x22 // "
-        static let Escape: UInt8         = 0x5C // \
+struct JSONParser {
+    init() {}
+
+    @inlinable
+    func parse<Bytes: Collection>(bytes: Bytes) throws
+        -> JSONValue where Bytes.Element == UInt8
+    {
+        var impl = JSONParserImpl(bytes: bytes)
```

Comment:
Is there a benefit to having `JSONParserImpl`, could its code not just be folded into `JSONParser`?. Then it can be invoked as:
```
let parser = JSONParser(bytes: ...)
try parser.parse()
```

Reply:
Fixed.